XenModel/Actions/SupplementalPack/UploadSupplementalPackAction.cs

```diff
@@ -188,6 +188,8 @@ private string UploadSupplementalPack(SR sr)
                 {
                     var poolUpdateRef = Pool_update.introduce(Connection.Session, vdiRef);
                     poolUpdate = Connection.WaitForCache(poolUpdateRef);
+

```

Comment:
I think the action should throw an exception if the poolUpdate is null (an update hasn't been retrieved from the cache)